snuba/querylog/__init__.py

```diff
@@ -13,7 +13,10 @@
 
 
 def record_query(
-    request: Request, timer: Timer, query_metadata: SnubaQueryMetadata
+    request: Request,
+    timer: Timer,
+    query_metadata: SnubaQueryMetadata,
+    extra_data: Mapping[str, Any],
```

Comment:
How is this going to work for split queries? In those cases there will be multiple independent clickhoouse queries each with its own experiments.

Reply:
It won't work correctly. We'll tag the experiments from both queries on the same transaction. I'm not trying to solve that in this particular PR. I'm actually not sure how to solve that without splitting out separate transactions for each query.

Reply:
I don't think we should have multiple transactions. A transaction for a backend service should be a request-response cycle. 
Well, the extra data column is already broken for split queries so I guess it is equally ok-ish ?